src/DefinitionStorage.php

```diff
@@ -96,12 +96,33 @@ public function set(string $id, mixed $definition): void
      *
      * @throws CircularReferenceException
      */
-    private function isResolvable(string $id, array $building): bool
+    private function isResolvable(string $id, array $building, ?string $parameterName = null): bool
     {
         if (isset($this->definitions[$id])) {
             return true;
         }
 
+        if (
+            $parameterName !== null
+            && isset($this->definitions[$id . '$' . $parameterName])
```

Comment:
Seems. this code added conflicts. For exmaple id "my$var" and id "my" for parameter "var". Need tests.

Reply:
It could be